src/nunavut/lang/cpp/templates/_fields.j2

```diff
@@ -11,4 +11,7 @@
     {% endif -%}
     {{ field.doc | block_comment('cpp-doxygen', 4, 120) }}
     {{ field.data_type | declaration }} {{ field | id }};
+    {%- if field.data_type is ArrayType %}
+    static constexpr {{ typename_unsigned_length }} {{ field | id }}_ArrayCapacity = {{ field.data_type.capacity }}U;
```

Comment:
When programming C++ we should not degrade to C. This is a C-like solution and isn't consistent with using the rest of the C++ types. How about we define something like this:
```
template <typename T, std::size_t MaxSize, typename Allocator = std::allocator<T>>
class VariableLengthArrayAsVector : public std::vector<T, Allocator>
{
public:
    VariableLengthArrayAsVector(std::initializer_list<T> l) :
       std::vector<T, Allocator>(l)
    {}

    static constexpr const std::size_t max_size = MaxSize;
};
```

While this has the advantage of allowing consumers to treat VariableLengthArrayAsVector types as a plan `std::vector` it does mean we have to adapt our baseclass constructors and template parameters to be in-sync with vector (for example we'd need to change the default allocator to `std::pmr::polymorphic_allocator` if generating c++17 or above).

Reply:
Another, similar option is to generate a type for each of these fields:
```
struct FooType
{
    struct BarType : public std::vector<std::uint8_t> {
        static constexpr const std::size_t max_size = 4;
    } bar;
};
```

Where the type is \<Field Name\> + "Type"

Reply:
Do we really need to implement all the constructors? Can't we say

```cpp
template <typename T, std::size_t MaxSize, typename Allocator = std::allocator<T>>
class VariableLengthArrayAsVector : public std::vector<T, Allocator>
{
public:
    using std::vector<T, Allocator>::vector;

    static constexpr const std::size_t max_size = MaxSize;
};
```

So all the `std::vector` constructors will be inherited?

I like this approach. Maybe we can make this pluggable? So the user can choose which VLA type they want from either a predefined list or a custom value?
As of right now, there are several variants/:

1. plain `std::vector` w/o any max size information
2. support library type based on `std::vector`
3. custom support library type
4. external type

Reply:
I think we can add/modify the following options:

* `variable_lenght_array_template` - a `str`, defaults to `std::vector`
* `variable_lenght_array_template_header` - header name, defaults to `<vector>`
* `variable_lenght_array_extra_template_arguments` - a list of str, defaults to empty list

So the full type would be:
```jinja

{{ variable_lenght_array_template}<{field| type}}{%for arg in variable_lenght_array_extra_template_arguments%},{arg}{% endif %}>
```

Reply:
See https://github.com/OpenCyphal/nunavut/blob/main/src/nunavut/lang/properties.yaml#L315 . My hope is we can tweak this property to do what we need.

Reply:
> Do we have any use cases in mind where such reconfigurability would actually be useful? Or is it a case of YAGNI?

Given that variable length arrays (and strings) is a significant source of pain for embedded systems I think providing a couple of bad defaults (i.e. vector using dynamic memory and VariableLengthArray being a barely serviceable but deterministic implementation) requires us to also allow the user to provide a good override that is optimal for their system.